tests/rt.sh

```diff
@@ -320,9 +324,12 @@ elif [[ $MACHINE_ID = orion ]]; then
   ROCOTORUN=$(which rocotorun)
   ROCOTOSTAT=$(which rocotostat)
   ROCOTOCOMPLETE=$(which rocotocomplete)
-  export PATH=/work/noaa/nems/emc.nemspara/soft/miniconda3/bin:$PATH
-  export PYTHONPATH=/work/noaa/nems/emc.nemspara/soft/miniconda3/lib/python3.8/site-packages
-  ECFLOW_START=/work/noaa/nems/emc.nemspara/soft/miniconda3/bin/ecflow_start.sh
+  export PATH=/work/noaa/da/role-da/spack-stack/miniconda-3.9.7/bin:$PATH
```

Comment:
Is DA group OK to support future miniconda package for UFS weather model?

Reply:
This is I believe the JCSDA maintained spack-stack, probably want to ask @climbfuji

Reply:
I have a general question, why do we use modules from different spack stacks? I find it very confusing, and potentially error prone if different stacks are compiled differently, with different compilers, mpi libraries etc. Isn't this PR supposed to use a unified  spack environment? 

For example on Hera:
```
prepend_path("MODULEPATH", "/scratch1/NCEPDEV/nems/role.epic/spack-stack/spack-stack-1.4.1/envs/ufs-pio-2.5.10/install/modulefiles/Core")
prepend_path("MODULEPATH", "/scratch1/NCEPDEV/jcsda/jedipara/spack-stack/modulefiles")
```
Why do we need two spack-stacks? Does one depend on the other? Does order of loading matter?

Reply:
These are not separate stacks in the sense that they are not full spack-stack installations. There are a few dependencies that are difficult to provide through Spack itself, so we have a few external dependencies contained in that second directory structure. This "second environment" will not create conflicts.

Reply:
@junwang-noaa from previous discussions with @climbfuji, i believe the goal is to eventually move away from organizational installs of miniconda/python and utilize sys. admin installed versions (for example, we did this for hercules). however, this will probably not happen for a while

Reply:
@DusanJovic-NOAA in general, `ecflow`, `mysql`, and `qt` are not built with spack or as part of any versioned spack-stack release/installation. this is due to difficulties in building these aforementioned packages using spack, as @AlexanderRichert-NOAA mentioned. please see https://spack-stack.readthedocs.io/en/latest/MaintainersSection.html#ecflow-with-gui-and-python or https://spack-stack.readthedocs.io/en/latest/MaintainersSection.html#mysql-server-and-client for more information.

that said, paths such as `/scratch1/NCEPDEV/jcsda/jedipara/spack-stack/modulefiles` generally contain only the modulefiles for those externally installed packages (i.e., `ecflow`, `mysql`, and often `miniconda`). these are used for all versioned spack-stack releases and are not "stacks" whatsoever, merely packages that spack-stack installations require.

Reply:
But ufs-weather-model (the model itself, the executable) does not need ecflow, mysql, miniconda or qt. ecflow is needed by rt.sh, and rt.sh loads the ecflow module. No need to load it from ufs_hera.intel.lua. During the compilation python is required but it is available in /usr/bin/python3.

I think, on hera specifically, this can be removed:

```diff
 diff --git a/modulefiles/ufs_hera.intel.lua b/modulefiles/ufs_hera.intel.lua
index 0e90fc79..dfed603a 100644
--- a/modulefiles/ufs_hera.intel.lua
+++ b/modulefiles/ufs_hera.intel.lua
@@ -3,7 +3,6 @@ loads UFS Model prerequisites for Hera/Intel
 ]])
 
 prepend_path("MODULEPATH", "/scratch1/NCEPDEV/nems/role.epic/spack-stack/spack-stack-1.4.1/envs/ufs-pio-2.5.10/install/modulefiles/Core")
-prepend_path("MODULEPATH", "/scratch1/NCEPDEV/jcsda/jedipara/spack-stack/modulefiles")
 
 stack_intel_ver=os.getenv("stack_intel_ver") or "2021.5.0"
 load(pathJoin("stack-intel", stack_intel_ver))
@@ -11,9 +10,6 @@ load(pathJoin("stack-intel", stack_intel_ver))
 stack_impi_ver=os.getenv("stack_impi_ver") or "2021.5.1"
 load(pathJoin("stack-intel-oneapi-mpi", stack_impi_ver))
 
-stack_python_ver=os.getenv("stack_python_ver") or "3.9.12"
-load(pathJoin("stack-python", stack_python_ver))
-
 cmake_ver=os.getenv("cmake_ver") or "3.23.1"
 load(pathJoin("cmake", cmake_ver))
 
diff --git a/tests/rt.sh b/tests/rt.sh
index 1cfea90a..793dbac9 100755
--- a/tests/rt.sh
+++ b/tests/rt.sh
@@ -296,10 +296,6 @@ elif [[ $MACHINE_ID = hera ]]; then
   ROCOTOCOMPLETE=$(which rocotocomplete)
   ROCOTO_SCHEDULER=slurm
 
-  PYTHONHOME=/scratch1/NCEPDEV/jcsda/jedipara/spack-stack/miniconda-3.9.12
-  export PATH=$PYTHONHOME/bin:$PATH
-  export PYTHONPATH=$PYTHONHOME/lib/python3.9/site-packages
-  
   module load ecflow/5.5.3
   ECFLOW_START=ecflow_start.sh
```

Reply:
Looking at this supplemental installation directory on Hera, for instance, it's got miniconda, OpenMPI, and MySQL (Qt is there too but not used, as far as I can tell). I'm not a huge fan of this approach, with the possible exception of MySQL which has some very annoying dependencies (specifically, each version of MySQL is built to work with one and only one specific version of Boost libs...). But I don't know that it's impossible to manage, either.

As far as @DusanJovic-NOAA's point about removing it on Hera, I'm fine with that as long as everything still runs. As I recall, the python needed for building UFS doesn't need any special features/modules, and ecFlow is coming from the common installation anyway. I don't think we'd be able to remove it from ufs_hera.gnu.lua on account of OpenMPI being installed there (which to be honest I'm not sure why that's installed outside of spack-stack).

Reply:
@DusanJovic-NOAA we include
```
prepend_path("MODULEPATH", "/scratch1/NCEPDEV/jcsda/jedipara/spack-stack/modulefiles")

stack_python_ver=os.getenv("stack_python_ver") or "3.9.12"
-load(pathJoin("stack-python", stack_python_ver))
```  

to avoid library conflicts between the python used to build the spack-stack itself and other external python versions (such as `/usr/bin/python3`). @AlexanderRichert-NOAA if you think this can be avoided, i am open to removing it, but i have seen mixing the spack python and other python installations cause problems before (e.g. within the srw).

Reply:
Does any of the libraries that we load from the spack-stack installation, specifically those listed in ufs_common, need or use python or the python packages installed in spack-stack? The libraries I'm familiar with do not.

Reply:
There used to be compare_ncfile.py or whatever it was called which used py-netcdf4, but we're taking that out anyway. That's the only thing that comes to mind; I don't think anything is using numpy, scipy, pandas, python hdf5/netcdf, ...

Reply:
I think if there's anything else needed to be discussed here it can happen port PR in an e-mail if needed. Resolving.